examples/with-datadog-client/src/api-client-helpers.ts

```diff
@@ -0,0 +1,138 @@
+import * as datadogAPIClient from '@datadog/datadog-api-client';
+import {
+    Configuration,
+    HttpLibrary,
+    RequestContext,
+    ResponseBody,
+    ResponseContext
+} from '@datadog/datadog-api-client/dist/packages/datadog-api-client-v1';
+import * as uiExtensionsSDK from '@datadog/ui-extensions-sdk';
+
+/**
+ * We attempt to convert from what the API client has for HTTP methods,
+ * to what the SDK uses for HTTP methods.
+ * This is a lossy conversion since the SDK doesn't support all HTTP methods.
+ *
+ * @param method The API client method to parse.
+ */
+function parseHTTPMethod(
+    method: datadogAPIClient.v1.HttpMethod
+): uiExtensionsSDK.ApiRequestMethod | undefined {
+    switch (method) {
+        case datadogAPIClient.v1.HttpMethod.CONNECT:
+            return;
+        case datadogAPIClient.v1.HttpMethod.DELETE:
+            return 'DELETE';
+        case datadogAPIClient.v1.HttpMethod.GET:
+            return 'GET';
+        case datadogAPIClient.v1.HttpMethod.HEAD:
+            return;
+        case datadogAPIClient.v1.HttpMethod.OPTIONS:
+            return;
+        case datadogAPIClient.v1.HttpMethod.PATCH:
+            return 'PATCH';
+        case datadogAPIClient.v1.HttpMethod.POST:
+            return 'POST';
+        case datadogAPIClient.v1.HttpMethod.PUT:
+            return 'PUT';
+        case datadogAPIClient.v1.HttpMethod.TRACE:
+            return;
+    }
+}
+
+/**
+ * We convert from whatever response we get from the SDK,
+ * to what the API client expects.
+ *
+ * @param response Anything coming from the SDK.
+ */
+function parseResponseBody(response: unknown): ResponseBody {
+    return {
+        binary: async (): Promise<Buffer> => {
+            // TODO: Converting to a string just to convert to a `Buffer` seems not quite right.
+            return new Buffer(JSON.stringify(response));
+        },
```

Comment:
This straight up might not work, because of postMessage serialization. We'll have to experiment